metadata/metadata.go

```diff
@@ -90,6 +90,20 @@ func Pairs(kv ...string) MD {
 	return md
 }
 
+// String implements the Stringer interface for pretty-printing a MD. Ordering of the values is non-deterministic as it ranges over a map.
+func (md MD) String() string {
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "MD{")
+	for k, v := range md {
+		if sb.Len() > 3 {
+			fmt.Fprintf(&sb, ", ")
+		}
```

Comment:
Not sure why do we need this conditional, removing this still makes the tests pass

Reply:
This is to avoid having a last trailing `","`, so we have `"MD{k1=[v1]}"` instead of `"MD{k1=[v1,]}"`.

Reply:
I'm still confused. Isn't that logic already done in `strings.Join(v, ", ")` below?

Reply:
thanks for clarifying.